Feature/450 org membership requirement for routes

OButlerOcc opened the pull request:
Adds a route guard and adjusts controlls based on if the signed in user is a member of an organisation.
Clicking a link wile not being a member redirect s to the request page.

Supported by:

https://github.com/MauroDataMapper/sde-resources/pull/244

https://github.com/MauroDataMapper/sde-core/pull/343

closes #450

OButlerOcc committed:
gh-#450-working commit
Modified the login to set a local cache flag saying weather a user has organisation membership

added a guard to prevent access to routes barred by not having membership

Add a class and check to hide header controlls based on set flags

OButlerOcc committed:
gh -450 - removed duplicated code in userService

OButlerOcc committed:
Test fixes

OButlerOcc committed:
gh-450-add licence

NigelPalmer reviewed (changes requested):
The Requests menu option has disappeared when you log in, even if you are a member of an organisation.
![image](https://github.com/user-attachments/assets/cc31a742-0352-4161-915d-c75f91913e76)

OButlerOcc committed:
gh-450, fixed some bugs present in the sign in/sign out logic. fixed the refresh logic

OButlerOcc commented:
> The Requests menu option has disappeared when you log in, even if you are a member of an organisation. ![image](https://private-user-images.githubusercontent.com/110250803/410414404-cc31a742-0352-4161-915d-c75f91913e76.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg5Mzk4MzMsIm5iZiI6MTczODkzOTUzMywicGF0aCI6Ii8xMTAyNTA4MDMvNDEwNDE0NDA0LWNjMzFhNzQyLTAzNTItNDE2MS05MTVkLWM3NWY5MTkxM2U3Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIwN1QxNDQ1MzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NTU2NjlhMjBlMjkwNDE4NzYxYTMyZjM0ODYyYTMxMTg5YWFhNzAzMjY2ZTVlYmMxMzM5NDI1YWYwNzc2ZmZhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.UCv95DkdZ4DZgUsrMO_W1FewYy-x8egmSM5tobTEXVQ)

This was a symptom of a screw up with the flags, Good thing you caught it before the demo.
Modified the flags to actually work correctly.
Corrected a typo checking for the wrong saved variable.
Added flag checks and saves so it now works when the page is reloaded and sign out correctly clears.

OButlerOcc committed:
gh-450 a test was broken in merge

mhocc reviewed (approved):
This all appears to be working as required, will probably need `sde-resources#237` merging first to make the build errors go away.

NigelPalmer committed:
Merge branch 'develop' into feature/450-org-membership-requirement-for-routes
